src/main/java/org/wso2/carbon/inbound/kafka/KafkaMessageConsumer.java

```diff
@@ -130,25 +133,44 @@ private void commitRecords(ConsumerRecords<byte[], byte[]> records) {
                 // Fixes - CS0196510
                 Object value  = record.value();
                 boolean isConsumed = false;
+                boolean poisonPillDetected = false;
                 if ( value == null ) {
+                    if (isPoisonPill(record)) {
```

Comment:
Shouldn't we have this poison check for records which has a value?

Reply:
No, for poison messages the value will be always null.